src/CMakeLists.txt

```diff
@@ -17,23 +17,32 @@ endif ()
 
 # compiler flags
 # TODO: separate GCC and Clang warnings; add more
-set(
-  CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
-  -Wall \
-  -Wextra \
-  -Wshadow \
-  -Wformat-nonliteral \
-  -Wformat-security \
-  -Wnon-virtual-dtor \
-  ")
-
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
-
-set(sanitizer_flags "-fsanitize=address -fsanitize=undefined")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${sanitizer_flags}")
-set(CMAKE_MODULE_LINKER_FLAGS_DEBUG
-    "${CMAKE_MODULE_LINKER_FLAGS_DEBUG} ${sanitizer_flags}")
-
+if (WIN32)
+  # compiler flags for MSVC
+  set(
+    CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} \
+    /W4 \
+    /wd4100 /wd4458 /wd4127 /wd4244 /wd4267 /wd4305 /wd4800 \
+    ")
```

Comment:
Could you document what these are somehow?

Also, it might be better to use [`add_compile_options`](https://cmake.org/cmake/help/latest/command/add_compile_options.html), instead of setting `CMAKE_CXX_FLAGS` directly. I know that i didn't do that below, but I was pretty new to cmake at the time.

Reply:
/W4 in MSVC has the same effect as -Wall -Wextra ... in GCC/Clang.
/wdnnnn is my Misuse. it suppresses some compiler warnings, which is not our purpose. I will remove it later

[MSVC Warning level:](https://learn.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=msvc-170)
>/W4 displays level 1, level 2, and level 3 warnings, and all level 4 (informational) warnings that aren't off by default. We recommend that you use this option to provide lint-like warnings. For a new project, it may be best to use /W4 in all compilations. This option helps ensure the fewest possible hard-to-find code defects.

> /wdnnnn	Suppresses the compiler warning that is specified by nnnn.
> For example, /wd4326 suppresses compiler warning C4326.